workflow/validate/validate.go

```diff
@@ -750,6 +749,9 @@ func (ctx *templateValidationCtx) validateSteps(scope map[string]interface{}, tm
 			if err != nil {
 				return errors.Errorf(errors.CodeBadRequest, "templates.%s.steps[%d].%s %s", tmpl.Name, i, step.Name, err.Error())
 			}
+			if step.ExitTemplate != nil {
```

Comment:
Why guard this statement?

Reply:
current template variable reference is only applicable in `exitTemplate`.